### PR TITLE
[API-37988] POA v1 POST /2122 - Use Benefits Documents API

### PR DESCRIPTION
**Pull request by tycol7**
## Summary

- *This work is behind a feature toggle (flipper): YES*
- Add feature flag to the POA controller.
  - If the feature flag is enabled, use Benefits Documents API upload
  - Otherwise, continue using VBMS

## Related issue(s)

[API-37988](https://jira.devops.va.gov/browse/API-37988)

## Testing done

- [x] *New code is covered by unit tests*

## Screenshots
N/A

## What areas of the site does it impact?
Benefits Documents upload; POA v1 POST

## Acceptance criteria

- [x]  I added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

Could use feedback on BD API restructure and testing approach.

### CODE REVIEW

File: modules/claims_api/app/sidekiq/claims_api/poa_updater.rb [hunk 1]
```diff
@@ -28,7 +28,9 @@ def perform(power_of_attorney_id) # rubocop:disable Metrics/MethodLength
 
         ClaimsApi::Logger.log('poa', poa_id: poa_form.id, detail: 'BIRLS Success')
 
-        ClaimsApi::PoaVBMSUpdater.perform_async(poa_form.id) if enable_vbms_access?(poa_form:)
+        if enable_vbms_access?(poa_form:) && !Flipper.enabled?(:lighthouse_claims_api_poa_use_bd)
```
Comment (FonzMP) [inline modules/claims_api/app/sidekiq/claims_api/poa_updater.rb:31]: this flipper check shouldn't be added here - we'll want the vbms updater to run either way, we just want to switch the ability to submit to VBMS <> Benefits Documents, so lines 27-32 in the poa_form_builder_job

this job still needs to run on every request if enable_vbms_access?

Comment (tycol7) [reply modules/claims_api/app/sidekiq/claims_api/poa_updater.rb]: https://github.com/department-of-veterans-affairs/vets-api/pull/17587/commits/e56deb8c19f7b0094c7f189c4200e77d1b2eb966

File: modules/claims_api/spec/sidekiq/poa_form_builder_job_spec.rb [hunk 2]
```diff
@@ -170,4 +171,24 @@
...
+
+  context 'when the BD upload feature flag is enabled' do
+    it 'uploads to the benefits document API' do
+      Flipper.enable(:lighthouse_claims_api_poa_use_bd)
+      expect(ClaimsApi::VBMSUploader).not_to receive(:new)
+      expect_any_instance_of(ClaimsApi::VBMSUploader).not_to receive(:upload_document)
+      power_of_attorney.update(form_data: power_of_attorney.form_data.deep_merge(
+        {
+          docType: 'L190'
+        }
```
Comment (tycol7) [inline modules/claims_api/spec/sidekiq/poa_form_builder_job_spec.rb:183]: From the task description:

> Uploads of VA Form 21-22, Appointment of Veterans Service Organization as Claimant's Representative, will include docType of L190

Can we assume the `docType` will be at the top level of the request body?

Comment (tycol7) [reply modules/claims_api/spec/sidekiq/poa_form_builder_job_spec.rb]: https://github.com/department-of-veterans-affairs/vets-api/pull/17587/commits/987270ad204d288917a1149f4bdc0c476e9b3749

File: modules/claims_api/lib/bd/bd.rb [hunk 3]
```diff
@@ -74,13 +100,14 @@ def upload(claim:, pdf_path:, pctpnt_vet_id: nil, doc_type: 'L122', file_number:
...
       auth_headers = claim.auth_headers
-      veteran_name = "#{auth_headers['va_eauth_firstName']}_#{auth_headers['va_eauth_lastName']}"
+      veteran_name = compact_veteran_name(auth_headers['va_eauth_firstName'], auth_headers['va_eauth_lastName'])
       birls_file_num = auth_headers['va_eauth_birlsfilenumber'] || file_number if doc_type != 'L705'
-      claim_id = doc_type == 'L705' ? claim.claim_id : claim.evss_id
+      claim_id = get_claim_id(doc_type, claim)
       file_name = generate_file_name(doc_type:, veteran_name:, claim_id:, original_filename:)
-      participant_id = pctpnt_vet_id if doc_type == 'L705'
+      participant_id = pctpnt_vet_id if %w[L075 L190 L750].include?(doc_type)
+      system_name = 'LH-B' if %w[L075 L190].include?(doc_type)
```
Comment (FonzMP) [inline modules/claims_api/lib/bd/bd.rb:108]: I believe we're on hold for this ticket as 'LH-B' was not a potential for submitters. We need to either a.) choose from the list returned or b.) make a decision of which of the available values we'll use. 

awaiting return of @klwatanabe

Comment (klwatanabe) [reply modules/claims_api/lib/bd/bd.rb:108]: @FonzMP, what do you mean by 'LH-B' was not a potential for submitters? I thought we were sending 'LH-B' today for VBMS uploads. Are you saying that the Documents API won't accept that?

Comment (klwatanabe) [reply modules/claims_api/lib/bd/bd.rb:108]: Oh, sorry, missed the comment on the ticket where @tycol7 noted the returned error from Benefits Documents API.
`Invalid content Source 'LH-B'. Expected one of VBMS, Lighthouse... `

We should double check with team Firefly to confirm that we should use 'Lighthouse' vs 'LH-B'. The goal is to be able to track that the document was updated by Lighthouse. Our team may have been the only ones using 'LH-B' if 'Lighthouse' is available now. 

If we do go that route, we should confirm how it is displayed in VBMS UAT for those uploads.

Comment (tycol7) [reply modules/claims_api/lib/bd/bd.rb:108]: Per Slack, go with `Lighthouse` for now (to test and re-evaluate if needed).

Comment (tycol7) [reply modules/claims_api/lib/bd/bd.rb]: https://github.com/department-of-veterans-affairs/vets-api/pull/17587/commits/8a1c09075b51c3a19d56b4e2c19ab6f4255f1b66

File: modules/claims_api/lib/bd/bd.rb [hunk 4]
```diff
@@ -74,13 +100,14 @@ def upload(claim:, pdf_path:, pctpnt_vet_id: nil, doc_type: 'L122', file_number:
...
       payload = {}
       auth_headers = claim.auth_headers
-      veteran_name = "#{auth_headers['va_eauth_firstName']}_#{auth_headers['va_eauth_lastName']}"
+      veteran_name = compact_veteran_name(auth_headers['va_eauth_firstName'], auth_headers['va_eauth_lastName'])
       birls_file_num = auth_headers['va_eauth_birlsfilenumber'] || file_number if doc_type != 'L705'
-      claim_id = doc_type == 'L705' ? claim.claim_id : claim.evss_id
+      claim_id = get_claim_id(doc_type, claim)
       file_name = generate_file_name(doc_type:, veteran_name:, claim_id:, original_filename:)
-      participant_id = pctpnt_vet_id if doc_type == 'L705'
+      participant_id = pctpnt_vet_id if %w[L075 L190 L750].include?(doc_type)
```
Comment (FonzMP) [inline modules/claims_api/lib/bd/bd.rb:107]: formerly L705 - assuming this is a typo.

Comment (tycol7) [reply modules/claims_api/lib/bd/bd.rb]: https://github.com/department-of-veterans-affairs/vets-api/pull/17587/commits/5da8c4d5f92c5136a2c7eb85225fe7db28e5a9fb

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**